src/main/kotlin/org/rust/lang/core/resolve/ref/RsMacroBodyReferenceDelegateImpl.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.ResolveResult
+import org.rust.lang.core.macros.findExpansionElements
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsWeakReferenceElement
+import org.rust.lang.core.psi.ext.ancestors
+import org.rust.lang.core.psi.ext.startOffset
+
+class RsMacroBodyReferenceDelegateImpl(
+    element: RsWeakReferenceElement
+) : RsReferenceBase<RsWeakReferenceElement>(element) {
+    override val RsWeakReferenceElement.referenceAnchor: PsiElement?
+        get() = element.referenceNameElement
+
+    private val delegates: List<RsReference>
+        get() {
+            return element.findExpansionElements()?.mapNotNull { delegated ->
+                val reference = delegated.ancestors
+                    .mapNotNull { it.reference }
+                    .firstOrNull() ?: return@mapNotNull null
+
+                val rangeInElement = TextRange.from(
+                    delegated.startOffset - reference.element.textRange.startOffset,
+                    delegated.textLength
+                )
+                if (rangeInElement != reference.rangeInElement) return@mapNotNull null
```

Comment:
Do you know case when it's not true?
It would be great to have it at least in comment to this code